libs/gi/ui/src/components/character/CharacterMultiSelectionModal.tsx

```diff
@@ -0,0 +1,500 @@
+              <WeaponToggle
+                onChange={(weaponType) =>
+                  database.displayCharacter.set({ weaponType })
+                }
+                value={weaponType}
+                totals={weaponTotals}
+                size="small"
+              />
+              <ElementToggle
+                onChange={(element) =>
+                  database.displayCharacter.set({ element })
+                }
+                value={element}
+                totals={elementTotals}
+                size="small"
+              />
+            </Box>
+            <IconButton
+              sx={{ ml: 'auto' }}
+              onClick={() => {
+                setSearchTerm('')
+                onHide()
+              }}
+            >
+              <CloseIcon />
+            </IconButton>
+          </Box>
+
+          <Box display="flex" gap={1}>
+            <TextField
+              autoFocus
+              value={searchTerm}
+              onChange={(e: ChangeEvent<HTMLTextAreaElement>) =>
+                setSearchTerm(e.target.value)
+              }
+              label={t('characterName')}
+              size="small"
+              sx={{ height: '100%', mr: 'auto' }}
+              InputProps={{
+                sx: { height: '100%' },
+              }}
+            />
+            <SortByButton
+              sortKeys={sortKeys}
+              value={sortType}
+              onChange={(sortType) =>
+                database.displayCharacter.set({ sortType })
+              }
+              ascending={ascending}
+              onChangeAsc={(ascending) =>
+                database.displayCharacter.set({ ascending })
+              }
+            />
+          </Box>
+        </CardContent>
+        <Divider />
+        <DataContext.Provider value={{ teamData: undefined } as any}>
+          <CardContent sx={{ flex: '1', overflow: 'auto' }}>
+            <Grid
+              container
+              spacing={1}
+              columns={{ xs: 2, sm: 3, md: 4, lg: 5 }}
+            >
+              {characterKeyList.map((characterKey) => (
+                <Grid item key={characterKey} xs={1}>
+                  <SelectionCard
+                    characterKey={characterKey}
+                    onClick={() => onClick(characterKey)}
+                    selectedCharKeys={currentTeamCharKeys}
+                  />
+                </Grid>
+              ))}
+            </Grid>
+          </CardContent>
+        </DataContext.Provider>
+      </CardThemed>
+    </ModalWrapper>
+  )
+}
+
+const CustomTooltip = styled(({ className, ...props }: TooltipProps) => (
+  <Tooltip {...props} classes={{ popper: className }} />
+))({
+  [`& .${tooltipClasses.tooltip}`]: {
+    padding: 0,
+  },
+})
+
+function SelectionCard({
+  characterKey,
+  onClick,
+  selectedCharKeys,
+}: {
+  characterKey: CharacterKey
+  onClick: () => void
+  selectedCharKeys: (CharacterKey | '')[]
```

Comment:
I would honestly try to simplify this logic by passing in the `selectedIndex: null|number`, instead of the list.